### ISSUE TITLE
Yacht Name autocomplete functionality

### ISSUE THREAD

**Issue by iteles** [charterindex/application#59]
As an admin of the site
I would like the Yacht Name field to autocomplete when I start typing 
So that I can prompt users to search for yachts we have listed and make the user experience smoother for them

## Acceptance Criteria
+ [x] In the search bar on the landing page #19 , the 'Yacht Name' field provides a drop-down list of names when a user starts typing
+ [ ] **On desktop only:** (agreed in sprint 1 planning session) In the search bar on the landing page #19 , the 'Yacht Name' field provides further information on yachts next to their names in the auto-complete dropdown options (see https://github.com/charterindex/application/issues/59#issuecomment-377715509)
  + [ ] Yacht length appears in our contrast pink colour next to the name of the yacht
  + [ ] Yacht type 'Power yacht' or 'Sailing yacht' appears in our contrast pink colour next to the length of the yacht
+ [ ] If an option in the autocomplete is too long for the screen, this should drop to the next line
+ [ ] There needs to be a differentiation of one option from another in the drop-down, either through whitespace or lines in accordance with the UI patterns already established in the designs.
+ [x] Clicking on an autocomplete option selects it
+ [ ] Using arrow keys to navigate through the autocomplete drop-down and then pressing enter selects that option

@chriswebley I propose we model this on the functionality implemented for the Science Museum in the first instance (this is a common user pattern, similar to Amazon for example) and move on from there.
http://collection.sciencemuseum.org.uk/

<img width="1059" alt="screen shot 2018-03-31 at 19 34 46" src="https://user-images.githubusercontent.com/4185328/38166390-fbcd2758-351a-11e8-92b5-431758a2a3ab.png">

<hr/>

### Original issue questions (now answered)
- You mention an 'autocomplete' functionality on the 'Yacht Name' section which would reference a list of yacht names (current and old), is this a list you would provide us with or is this data all available from the API?
- How do I select an autocomplete option - by clicking on it or can I also press my down arrow key to navigate through the list and then press enter to select the one that is highlighted?

<img src="https://user-images.githubusercontent.com/16775804/38046237-0cd3c964-32b7-11e8-8505-39283f4abbb5.png" width=350px />

**Comment by chriswebley** [comment]
Yes we have a list of these in the DB. 2 fields 'yacht name' and 'Previous Yacht Names' which allows up to 2 previous yacht names to be in there. The maximum amount of yacht names for 1 yacht record could be 3 names.

I'd also like the previous yacht names drop down to show Yacht type and length next to the suggested criteria. Something like the Amazon bar (see this example)
![sample](https://user-images.githubusercontent.com/36406293/38166594-38fc3c2e-351e-11e8-9a0f-352cbfea45a1.jpg)

**Comment by chriswebley** [comment]
Clicking on an auto complete item or using the keyboard works. I'm happy for you to use which ever you feel is a better option. The ones I've seen and like do both.

**Comment by PeterTrotter** [comment]
@Cleop Searching by name should now be working on the API e.g. [api/yachts?filter[name]=na&page[number]=1&page[size]=15&fields[yachts]=name,urn,lowPrice](https://charterindex.charterindexdev.com/api/yachts?filter[name]=na&page[number]=1&page[size]=15&fields[yachts]=name,urn,lowPrice)

Default sort order is by relevance with yachts that start with the entered query appearing first followed by yachts containing the query. You can override this by requesting specific ordering. See #89

**Comment by SimonLab** [comment]
- investigating how to implement `debounce` to only send API request after a certain period of time instead of sending requests each time the input field change (avoid sending for example 20/30 requests for a search)
- Can we use https://github.com/thebritican/elm-autocomplete

**Comment by SimonLab** [comment]
- investigating how to implement `debounce` to only send API request after a certain period of time instead of sending requests each time the input field change (avoid sending for example 20/30 requests for a search)
- Can we use https://github.com/thebritican/elm-autocomplete

**Comment by SimonLab** [comment]
https://github.com/thebritican/elm-autocomplete is easy to work on a fixed list of items by defining your own filter function. However as we are using the API to get the correct element, hte list is dynamic so I don't think this package is the right one for our usage.
Instead we can reuse the structure of the other dropdown to display the list of yachts.

**Comment by SimonLab** [comment]
The autocomplete is now able to get information from the api:
![image](https://user-images.githubusercontent.com/6057298/38926730-9172c810-42fb-11e8-947c-6b1233a5f6f3.png)

Looking now at displaying the other information (length and yacht type)
